src/io/flutter/devtools/DevToolsManager.java

```diff
@@ -122,19 +132,63 @@ public void openBrowserAndConnect(String uri, String page) {
     openBrowserImpl(uri, page);
   }
 
+  /**
+   * Gets the process handler that will start DevTools from pub.
+   */
+  private OSProcessHandler getProcessHandlerForPub() {
+    final FlutterSdk sdk = FlutterSdk.getFlutterSdk(project);
+    if (sdk == null) {
+      return null;
+    }
+    // TODO(https://github.com/flutter/flutter/issues/33324): We shouldn't need a pubroot to call pub global.
+    @Nullable final PubRoot pubRoot = PubRoots.forProject(project).stream().findFirst().orElse(null);
+    final FlutterCommand command = sdk.flutterPackagesPub(pubRoot, "global", "run", "devtools", "--machine", "--port=0");
+
+    // TODO(devoncarew): Refactor this so that we don't use the console to display output - this steals
+    // focus away from the Run (or Debug) view.
+    return command.startInConsole(project);
+  }
+
+  /**
+   * Gets the process handler that wills tart DevTools in bazel.
+   */
+  private OSProcessHandler getProcessHandlerForBazel() {
+    final Workspace workspace = Workspace.load(project);
+    if (workspace == null || workspace.getDevtoolsScript() == null) {
+      return null;
+    }
+    final GeneralCommandLine commandLine = new GeneralCommandLine().withWorkDirectory(workspace.getRoot().getPath());
+    commandLine.setExePath(FileUtil.toSystemDependentName(workspace.getRoot().getPath() + "/" + workspace.getLaunchScript()));
+    commandLine.setCharset(CharsetToolkit.UTF8_CHARSET);
+    commandLine.addParameters(workspace.getDevtoolsScript(), "--", "--machine", "--port=0");
+    OSProcessHandler handler;
+    try {
+      handler = new OSProcessHandler(commandLine);
+    }
+    catch (ExecutionException e) {
+      e.printStackTrace();
```

Comment:
Should this be reported to users? Will returning null for the handler cause an issue to be reported for the failure to launch?

We should `Log.` this here, likely at the warn level.

Reply:
Done.